pkg/core/xds/types.go

```diff
@@ -188,6 +188,8 @@ type Proxy struct {
 	RuntimeExtensions map[string]interface{}
 	// Zone the zone the proxy is in
 	Zone string
+	// InternalAddresses is a set of address prefixes that are considered internal to the mesh, it will be configured to in Envoy HCM config
+	InternalAddresses []InternalAddress
```

Comment:
Adding this field on the `Proxy` object so that the value can be accessed by the generators and plugins. `InternalAddress` carry values of the same category of existing data like `ApiVersion`